src/D2L.Bmx/OktaAuthenticator.cs

```diff
@@ -57,18 +57,18 @@ bool ignoreCache
 		if( authnResponse is AuthenticateResponse.MfaRequired mfaInfo ) {
 			OktaMfaFactor mfaFactor = consolePrompter.SelectMfa( mfaInfo.Factors );
 
-			if( !IsMfaFactorTypeSupported( mfaFactor.FactorType ) ) {
+			if( mfaFactor.FactorName == "Unknown" ) {
 				throw new BmxException( "Selected MFA not supported by BMX" );
 			}
 
 			// TODO: Handle retry
-			if( mfaFactor.FactorType is "sms" or "call" or "email" ) {
+			if( mfaFactor.RequireChallengeIssue ) {
```

Comment:
Alternatively ``if ( mfaFactor is OktaMfaEmailFactor ... ) {``  etc....?

Reply:
as is seems good